laundry/views.py

```diff
@@ -0,0 +1,124 @@
+import calendar
+import datetime
+
+from django.utils import timezone
+from django.utils.timezone import make_aware
+from requests.exceptions import HTTPError
+from rest_framework import generics
+from rest_framework.response import Response
+
+from laundry.api_wrapper import Laundry
+from laundry.models import LaundrySnapshot
+from laundry.serializers import LaundrySnapshotSerializer
+
+
+laundry = Laundry()
+
+
+# should i just use the regular API view with a GET header?
+class Halls(generics.ListAPIView):
+    def list(self, request):
+        try:
+            return Response(laundry.all_status())
+        except HTTPError:
+            return Response({"error": "The laundry api is currently unavailable."})
+
+
+class HallInfo(generics.ListAPIView):
+    def list(self, request, hall_id):
+        try:
+            return Response(laundry.hall_status(int(hall_id)))
+        except ValueError:
+            return Response({"error": "Invalid hall id passed to server."})
+        except HTTPError:
+            return Response({"error": "The laundry api is currently unavailable."})
```

Comment:
These should return a non-200 status code. Probably a 404 for the ValueError and maybe a 503 for the HTTPError